R/query.R

```diff
@@ -190,25 +190,74 @@ orderly_find_name <- function(id, config, locate = FALSE, draft = TRUE,
   }
 }
 
+## This is only used in one place, and so we can be quite flexible in
+## how it is used.  It's a bit of a horror show, tbh, given how simple
+## what we want to achieve is.  This function is not an orderly API
+## function, and is used only in dependency resolution, so we can
+## update this later if needed.
 orderly_find_report <- function(id, name, config, locate = FALSE,
                                 draft = TRUE, must_work = FALSE) {
   config <- orderly_config_get(config, locate)
-  path <-
-    file.path((if (draft) path_draft else path_archive)(config$root), name)
-  if (id == "latest") {
-    id <- orderly_latest(name, config, FALSE,
-                         draft = draft, must_work = must_work)
+
+  if (is.character(draft)) {
+    ## draft <- match_value(draft, c("always", "newer", "never"))
+    search_draft <- draft != "never"
+    search_archive <- draft != "always"
+    what <- switch(draft,
+                   always = "draft",
+                   never = "archive",
+                   newer = "draft or archive")
+  } else {
+    search_draft <- draft
+    search_archive <- !draft
+    what <- if (draft) "draft" else "archive"
   }
-  path_report <- file.path(path, id)
-  if (!is.na(id) && file.exists(path_report)) {
-    return(path_report)
+
+  base_archive <- file.path(path_archive(config$root), name)
+  base_draft <- file.path(path_draft(config$root), name)
+
+  if (id == "latest") {
+    path <- NULL
+    if (search_archive) {
+      found <- orderly_latest(name, config, draft = FALSE, must_work = FALSE)
```

Comment:
Why don't we pass *must_work* to *orderly_latest* here (and below)?

Reply:
because either branch could fail (there could conceivably be no drafts but some archive reports) so the must_work is dealt with at the end of the function after trying (perhaps) both possibilities